### PR TITLE
Add a docs directory, and reorganize the docs a bit

### PR DESCRIPTION
**Pull request by perotinus**
I added a little bit of content, but the main intention here is to get a bit more structure and move content out of the top-level README.

cc @font, @pmorie, @madhusudancs

### CODE REVIEW

**Comment by perotinus** [comment]
@font, @pmorie, @madhusudancs Ping?

File: docs/development.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+regular basis, and some scripts/tooling in this project are built around Bazel.
+
+NOTE: There is an issue with bazel 0.6.x. As a workaround, use 0.5.x, or pass
+the flag `--incompatible_comprehension_variables_do_not_leak=false` to bazel
+0.6.x invocations.
+
+Before doing any development work, you must (in order, from the repository root
+directory, after cloning):
+
+1.  run `./update-codegen.sh`
```
Comment (ericchiang) [inline docs/development.md]: Might want to point out that you need to clone this to

```
$GOPATH/src/k8s.io/cluster-registry
```

And not

```
$GOPATH/src/github.com/kubernetes/cluster-registry
```

Also kubernetes/kubernetes commits these artifacts so non-bazel projects can import the APIs. Any thoughts there?

File: docs/development.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+1.  run `bazel run //:gazelle`
+
+## Building `clusterregistry`
+
+`clusterregistry` is the binary for the Kubernetes API server that serves the
+cluster registry API.
+
+To build it, from the root of the repository:
+
+1.  Run `bazel build //cmd/clusterregistry`
```
Comment (ericchiang) [inline docs/development.md]: (note: this might take a while the first time you run it...)

File: README.md [hunk 3]
```diff
@@ -1,51 +1,36 @@
...
+A lightweight tool for maintaining a list of clusters and associated metadata.
 
-This is currently in the prototyping stage, and is not yet ready for use except
-by contributors and experimenters.
+# What is it?
 
-If you have questions, please reach out to
-[kubernetes-sig-federation](https://groups.google.com/forum/#!forum/kubernetes-sig-federation).
+The cluster registry helps you keep track of and perform operations on all of
+your clusters. This repository contains an implementation of the cluster API=
```
Comment (ericchiang) [inline README.md]: `=`?

File: docs/development.md (hunk 1 above)
Comment (perotinus) [reply docs/development.md]: Good point. Though, interestingly enough, after #

Hmm. Personally I'd rather avoid committing artifacts until there are enough people asking for it.

Comment (ericchiang) [reply docs/development.md]: > Hmm. Personally I'd rather avoid committing artifacts until there are enough people asking for it.

works for me

**Comment by perotinus** [comment]
@ericchiang Yes, that failure is expected until #19 goes in. If you exclude `-//pkg/crinit/...`, it should work.

DO NOT MERGE until #18 and #19 go in.

File: docs/development.md (hunk 2 above)
Comment (perotinus) [reply docs/development.md]: Good point. Added a note to that effect.

File: README.md (hunk 3 above)
Comment (perotinus) [reply README.md]: That's not supposed to be there! Removed.

**Comment by perotinus** [comment]
Thanks for the review, @ericchiang!

File: README.md [hunk 4]
```diff
@@ -1,51 +1,36 @@
...
+[kubernetes-sig-multicluster](https://groups.google.com/forum/#!forum/kubernetes-sig-multicluster)
+mailing list, or find us on
+[Slack](https://github.com/kubernetes/community/blob/master/communication.md#social-media)
+in [#sig-federation](https://kubernetes.slack.com/messages/sig-federation/).
 
-## Building crinit
+## Maintainers
 
-From the root of the repository:
+-   [@perotinus](https://github.com/perotinus)
```
Comment (perotinus) [inline README.md]: @font Do you want to be added to this list?

**Comment by perotinus** [comment]
@font @madhusudancs @ericchiang If you don't have any further comments, I'm going to merge this today.

**Comment by ericchiang** [comment]
@perotinus did you address comments locally and not push the changes? I don't see any fixes for the review.

**Comment by perotinus** [comment]
@ericchiang I did exactly that. Changes are pushed now.

File: README.md (hunk 4 above)
Comment (font) [reply README.md]: I'm still ramping up but sure, thanks!

File: README.md [hunk 5]
```diff
@@ -1,51 +1,36 @@
...
+# Getting involved
 
-1.  run `update-codegen.sh`
-1.  run `bazel run //:gazelle`
+The cluster registry is still a young project, but we welcome your
+contributions, suggestions and input! Please reach out to the
+[kubernetes-sig-multicluster](https://groups.google.com/forum/#!forum/kubernetes-sig-multicluster)
+mailing list, or find us on
+[Slack](https://github.com/kubernetes/community/blob/master/communication.md#social-media)
+in [#sig-federation](https://kubernetes.slack.com/messages/sig-federation/).
```
Comment (font) [inline README.md]: Now `sig-multicluster`

File: docs/development.md (hunk 1 above)
Comment (font) [reply docs/development.md]: This script fails with:
```
generating clientset for group clusterregistry and version v1alpha1 at /home/ifont/go/src/k8s.io/cluster-registry/pkg/client
.............
INFO: Found 1 target...
Target @io_k8s_code_generator//cmd/client-gen:client-gen up-to-date:
  bazel-bin/external/io_k8s_code_generator/cmd/client-gen/client-gen
INFO: Elapsed time: 3.839s, Critical Path: 0.05s

INFO: Running command line: bazel-bin/external/io_k8s_code_generator/cmd/client-gen/client-gen --input-base k8s.io/cluster-registry/pkg/apis --input clusterregistry/ clusterregistry/v1alpha1 --clientset-path k8s.io/cluster-registry/pkg/client/clientset_generated '--output-base=/home/ifont/go/src'
ERROR: logging before flag.Parse: F1017 11:35:35.977724   16879 main.go:202] Error: Failed making a parser: unable to add directory "k8s.io/apimachinery/pkg/fields": unable to import "k8s.io/apimachinery/pkg/fields": cannot find package "k8s.io/apimachinery/pkg/fields" in any of:
        /usr/local/go/src/k8s.io/apimachinery/pkg/fields (from $GOROOT)
        /home/ifont/go/src/k8s.io/apimachinery/pkg/fields (from $GOPATH)
goroutine 1 [running]:
github.com/golang/glog.stacks(0xc42000e000, 0xc420118000, 0x17a, 0x1ce)
        external/com_github_golang_glog/glog.go:769 +0xa7
github.com/golang/glog.(*loggingT).output(0x8635c0, 0xc400000003, 0xc4200871e0, 0x84115f, 0x7, 0xca, 0x0)
        external/com_github_golang_glog/glog.go:720 +0x36b
github.com/golang/glog.(*loggingT).printf(0x8635c0, 0xc400000003, 0x6ffc79, 0x9, 0xc4200e5e70, 0x1, 0x1)
        external/com_github_golang_glog/glog.go:655 +0x14f
github.com/golang/glog.Fatalf(0x6ffc79, 0x9, 0xc4200e5e70, 0x1, 0x1)
        external/com_github_golang_glog/glog.go:1148 +0x67
main.main()
        external/io_k8s_code_generator/cmd/client-gen/main.go:202 +0x6c5
ERROR: Non-zero return code '255' from command: Process exited with status 255.
```
Unless you `go get k8s.io/apimachinery` first as a workaround. We appear to be missing this dependency.

File: README.md (hunk 5 above)
Comment (perotinus) [reply README.md]: Thanks, fixed

File: docs/development.md (hunk 1 above)
Comment (ericchiang) [reply docs/development.md]: @font I think you're hitting the issue https://github.com/kubernetes/cluster-registry/pull/18 addresses.

**Comment by perotinus** [comment]
Yes, the script currently fails for lack of k8s.io/apimachinery. #18 fixes that by using the version from the bazel workspace, so I'd rather wait for that PR to go in than document the workaround.

File: README.md [hunk 6]
```diff
@@ -1,51 +1,37 @@
...
-is an effort being led under the auspices of sig-multicluster.
+A lightweight tool for maintaining a list of clusters and associated metadata.
 
-This is currently in the prototyping stage, and is not yet ready for use except
-by contributors and experimenters.
+# What is it?
 
-If you have questions, please reach out to
-[kubernetes-sig-federation](https://groups.google.com/forum/#!forum/kubernetes-sig-federation).
+The cluster registry helps you keep track of and perform operations on all of
```
[the remark this answers is not in the document]
Comment (perotinus) [reply README.md]: Done.

File: README.md [hunk 7]
```diff
@@ -1,51 +1,37 @@
...
+mailing list, or find us on
+[Slack](https://github.com/kubernetes/community/blob/master/communication.md#social-media)
+in [#sig-multicluster](https://kubernetes.slack.com/messages/sig-multicluster/).
 
-## Building crinit
+## Maintainers
 
-From the root of the repository:
+-   [@perotinus](https://github.com/perotinus)
+-   [@font](https://github.com/font)
```
[the remark this answers is not in the document]
Comment (perotinus) [reply README.md]: Of course! Sorry, an oversight on my part.

File: docs/api_design.md [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# Cluster API design
+
+The API design for the cluster registry API is currently being discussed
+[here](https://docs.google.com/document/d/1Oi9EO3Jwtp69obakl-9YpLkP764GZzsz95XJlX1a960/edit).
```
[the remark this answers is not in the document]
Comment (perotinus) [reply docs/api_design.md]: Definitely, that was my plan. I'm planning to raise the PR at the end of this week.

**Comment by perotinus** [comment]
@font @madhusudancs @ericchiang Thanks for the reviews!

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (madhusudancs) [reply README.md]: No worries. Thanks for adding me.

File: docs/api_design.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (madhusudancs) [reply docs/api_design.md]: Cool, thanks.

**Pull request merged**